src/keybaseca/config/config.go

```diff
@@ -58,6 +58,35 @@ func ValidateConfig(conf EnvConfig, offline bool) error {
 			return fmt.Errorf("STRICT_LOGGING must be either 'true' or 'false', '%s' is not valid", conf.getStrictLogging())
 		}
 	}
+	if conf.GetKeybaseUsername() != "" || conf.GetKeybasePaperKey() != "" {
+		if conf.GetKeybaseUsername() == "" && conf.GetKeybasePaperKey() != "" {
+			return fmt.Errorf("you must set set a username if you set a paper key (username='%s', key='%s')", conf.GetKeybaseUsername(), conf.GetKeybasePaperKey())
+		}
+		if conf.GetKeybasePaperKey() == "" && conf.GetKeybaseUsername() != "" {
+			return fmt.Errorf("you must set set a paper key if you set a username (username='%s', key='%s')", conf.GetKeybaseUsername(), conf.GetKeybasePaperKey())
+		}
+		if !offline {
+			err := validateUsernamePaperkey(conf.GetKeybaseHomeDir(), conf.GetKeybaseUsername(), conf.GetKeybasePaperKey())
+			if err != nil {
+				return fmt.Errorf("failed to validate KEYBASE_USERNAME and KEYBASE_PAPERKEY: %v", err)
+			}
+		}
+	}
```

Comment:
you can probably reorder this big block if you dont want these nested if statements...
i think remove the outer one and then returning nil earlier if they're both blank. whatever. not a blocker.

Reply:
Hmm, but returning nil seems like it would cause issues if someone else was adding more validation later on after this `if` (which seems likely). So I think best to keep with the pattern of one big if for each thing that is being validated.

Reply:
Hmm, but returning nil seems like it would cause issues if more validation was added to the end of this function (which seems likely). So I think probably best to keep with this large if block.

Reply:
it's all a frame of reference. i could just as easily see the `validateUsernamePaperkey` call be at the bottom as the last thing you might consider doing in a flow of possible validations. no thang.